compiler/rustc_codegen_llvm/src/gotoc/mir_to_goto/codegen/place.rs

```diff
@@ -65,10 +65,18 @@ impl<'tcx> ProjectedPlace<'tcx> {
 
 /// Constructor
 impl<'tcx> ProjectedPlace<'tcx> {
-    fn check_expr_typ(expr: &Expr, typ: &TypeOrVariant<'tcx>, ctx: &mut GotocCtx<'tcx>) -> bool {
+    fn check_expr_typ_mismatch(
+        expr: &Expr,
+        typ: &TypeOrVariant<'tcx>,
+        ctx: &mut GotocCtx<'tcx>,
+    ) -> Option<(Type, Type)> {
         match typ {
-            TypeOrVariant::Type(t) => &ctx.codegen_ty(t) == expr.typ(),
-            TypeOrVariant::Variant(_) => true, //TODO, what to do here?
+            TypeOrVariant::Type(t) => {
+                let expr_ty = expr.typ().clone();
+                let type_from_mir = ctx.codegen_ty(t);
+                if expr_ty != type_from_mir { Some((expr_ty, type_from_mir)) } else { None }
+            }
+            TypeOrVariant::Variant(_) => None, //TODO, what to do here?
```

Comment:
Do we want to create an issue for this ? Not sure if this is tracked somewhere..

Reply:
Added